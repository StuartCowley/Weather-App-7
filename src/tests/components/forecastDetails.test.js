import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components /forecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 34,
    wind: {
      speed: 23,
      direction: "North West",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values of props correctly", () => {
    const { getByTestId } = render(<ForecastDetails forecast={validProps} />);

    expect(getByTestId("forecast-date")).toHaveClass("forecast-details_date");
    expect(getByTestId("forecast-temp-min")).toHaveClass(
      "forecast-details_temperatureMin"
    );
    expect(getByTestId("forecast-temp-max")).toHaveClass(
      "forecast-details_temperatureMax"
    );
    expect(getByTestId("forecast-humidity")).toHaveClass(
      "forecast-details_humidity"
    );
    expect(getByTestId("forecast-wind")).toHaveClass("forecast-details_wind");
  });
});
