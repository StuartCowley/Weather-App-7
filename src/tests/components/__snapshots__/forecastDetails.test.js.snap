// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details_date"
      data-testid="forecast-date"
    >
      Thu 4th Jan
    </div>
    <div
      class="forecast-details_temperatureMin"
      data-testid="forecast-temp-min"
    >
      Min temperature: 12 °C
    </div>
    <div
      class="forecast-details_temperatureMax"
      data-testid="forecast-temp-max"
    >
      Max temperature: 22 °C
    </div>
    <div
      class="forecast-details_wind"
      data-testid="forecast-wind"
    >
      Wind:23mph North West
    </div>
    <div
      class="forecast-details_humidity"
      data-testid="forecast-humidity"
    >
      Humidity: 34%
    </div>
  </div>
</DocumentFragment>
`;
